kpi-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java

```diff
@@ -0,0 +1,302 @@
+                    node.has("title") ? node.get("title").asText(null) : null,
+                    node.has("description") ? node.get("description").asText(null) : null,
+                    ctxt.readTreeAsValue(node.get("names"), ProfileNames.class),
+                    ctxt.readTreeAsValue(node.get("primarySource"), DataSourceSelection.class),
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("joins"), JoinSourceSelection[].class)),
+                    node.has("userId") ? node.get("userId").asText(null) : null,
+                    attributes,
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("attributeTags"), AttributeTag[].class))
+            );
+        }
+    }
+
+    protected <T> T getOrDefault(Supplier<T> function, T defaultValue) {
+        try {
+            T value = function.get();
+            if (value == null) {
+                return defaultValue;
+            }
+            return value;
+
+        } catch (NullPointerException | NotFoundException e) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public void run() {
+        var sessionExample = new SessionExample();
+        CortexSession cortexSession = sessionExample.getCortexSession();
+        RemoteStorageEnvLocator remoteStorageEnvLocator = new RemoteStorageEnvLocator(mapAsJavaMap(cortexSession.getContext().getSparkSession().conf().getAll()), new LocalRemoteStorageClient(null));
+        String bucketProtocol = remoteStorageEnvLocator.get().getProtocol();
+        String connectionType = remoteStorageEnvLocator.get().getType().toString();
+        Pair<String, String> bucketApiEndpoint = remoteStorageEnvLocator.get().getFeedbackAggregatorConfig();
+        String profilesBucket = remoteStorageEnvLocator.get().getBucketName("cortex-profiles");
+
+        String dataSourceConfig = "{\n" +
+                "                \"project\": \"" + project + "\",\n" +
+                "                \"attributes\": [\n" +
+                "                  \"timeOfExecution\",\n" +
+                "                  \"value\",\n" +
+                "                  \"startDate\",\n" +
+                "                  \"endDate\",\n" +
+                "                  \"windowDuration\"\n" +
+                "                ],\n" +
+                "                \"connection\": {\n" +
+                "                  \"name\": \"KPI-" + name + "\"\n" +
+                "                },\n" +
+                "                \"description\": \"" + description + "\",\n" +
+                "                \"kind\": \"batch\",\n" +
+                "                \"name\": \"KPI-" + name + "\",\n" +
+                "                \"primaryKey\": \"timeOfExecution\",\n" +
+                "                \"title\": \"" + name + "\"\n" +
+                "              }";
+
+        String connectionConfig = "{\n" +
+                "                \"project\": \"" + project + "\",\n" +
```

Comment:
Sucks that multiline strings were added in Java 15 and not earlier... https://stackoverflow.com/questions/878573/does-java-have-support-for-multiline-string

I kinda think that removing the stray newline characters might make things more readable, but I really don't know if this JSON can be formatted any nicer.